recipes/imagl/all/test_package/conanfile.py

```diff
@@ -7,7 +7,8 @@ class ImaglTestConan(ConanFile):
     generators = "cmake"
 
     def build(self):
-        cmake = CMake(self)
+        generator = "Ninja" if str(self.settings.compiler) == "Visual Studio" else None
+        cmake = CMake(self, generator=generator)
```

Comment:
I think using the default generator should be fine. No?
```suggestion
        cmake = CMake(self)
```

Reply:
The problem with default generator is that when you have multiple instance of MSVC (19.22 and 19.28 for example), it will always choose the last one. It will generate Visual Studio solution and project files, and use MSBuild to build it. I think it's a "feature" of CMake / Visual Studio generator that does not use environment which is set by vcvars.
So if you want to use a specific MSVC version, you have to use Ninja generator with CMake.

You can find some side explanation here: https://devblogs.microsoft.com/cppblog/side-by-side-minor-version-msvc-toolsets-in-visual-studio-2019/

Will Buik explains that there is some modification to do inside vcxproj files. I don't see how it is possible to do it in conan / cmake process. He then explains how to do it in CMake (from Visual Studio, but it is easy to have the same environment) by setting some environment variables and by using Ninja generator.

I'm sorry that it cannot be simple with MS.

So, the answer to "I think using the default generator should be fine. No?" is "No". 😢

Reply:
Thanks for the the link and explanation.
The article says that this would be fine only as a last resort (to work around bugs in your code or bugs in MSVC).
Is there an actual big this project is hitting?

Reply:
Not a bug, but some C++ features not implemented in old MSVC version.
imaGL requires minimal MSVC versions according to this table:
| imaGL version | MSVC minimal version |
|---------------|-----------------|
| 0.1.0 | 19.25 (default in VS 16.5) |
| 0.1.1 | 19.25 (default in VS 16.5) |
| 0.1.2 | 19.22 (default in VS 16.2) |
| 0.2.0 | 19.25 (default in VS 16.5) |
| 0.2.1 | 19.22 (default in VS 16.2) |

To test minimum version when building with newer VS (16.9+), we must install previous version according to Will Buik explanation. When Microsoft releases a new VS version, there is no way to install an older one. From today, there is only VS 16.9 available from MS. So to let's test with MSVC 19.22 or 19.25, I don't see any alternative.

Reply:
Okay then, can you then please add a comment about the how and why?
It would also be nice to print a message to `self.output` that this option is temporary/unstable.